packages/ui/src/common/OTPField.tsx

```diff
@@ -0,0 +1,68 @@
+import * as Common from '@clerk/elements/common';
+import React from 'react';
+
+import * as Field from '../primitives/field';
+
+export function OTPField({
+  label = 'One-time password',
+  ...props
+}: React.ComponentProps<typeof Common.Input> & {
+  label?: React.ReactNode;
+}) {
+  return (
+    <Common.Field
+      asChild
+      name='code'
+    >
+      <Field.Root>
+        <Common.Label asChild>
+          <Field.Label visuallyHidden>{label}</Field.Label>
+        </Common.Label>
+
+        <Common.FieldState>
+          {({ state }) => (
+            <Common.Input
+              type='otp'
+              autoSubmit
+              className='flex gap-2'
+              render={({ value, status }) => (
+                <Field.Input
+                  type='otp'
```

Comment:
I don't think we need this since this is being asChilded onto a span.

Reply:
We do in this case because it's how I'm setting the styles in the `Field.Input` component

Reply:
Can we name as `variant` or something else? Or if we're keen to keep as type make sure that type doesn't get added to the span?

Reply:
Sure!

Reply:
Done